tasks/internal/process-file-updates-task/process-file-updates-task.yaml

```diff
@@ -251,7 +251,66 @@ spec:
         fi
 
         echo -e "\n*** START LOCAL CHANGES ***\n"
-        git diff
+        # compare the differences between the working directory and the staging area 
+        git diff | tee "${TEMP}"/tempMRFile.diff
+
+        # compare the differences between the staging area and the latest commit
+        git diff --cached | tee "${TEMP}"/tempMRFile-cached.diff
+
+        if [[ -s "${TEMP}"/tempMRFile.diff ]]; then
+            # replacements exist 
+            # It's to deal with the lines like "@@ -N1,N2 +N3,N4 @@", 
+            # but it's possible to meet a line like "@@ -4,7 +4,7 @@ $schema: /app/app-settings.yml".
+            awk '/^@@/ {match($0, /@@ ([^@]+) @@/, arr); print arr[1]}' "${TEMP}"/tempMRFile.diff |\
+               tee "${TEMP}"/lineFile
+        elif [[ ! -s "${TEMP}"/tempMRFile-cached.diff ]]; then
+           # only seed exists and the file with the same content is there already 
+           echo "the seeding file already exists" \
+               | tee -a "$(results.fileUpdatesInfo.path)"
+           echo -n "Success" |tee "$(results.fileUpdatesState.path)"
+           exit 0
+        fi
+
+        openMRList=$(glab mr list -R "${UPSTREAM_REPO}" --search "Konflux release" |grep "^!"|| true)
```

Comment:
So this is the line where it failed. From Jing:

> When there is nothing returned in command `grep "^!" `, the return code is 1.  Then the script   exited in the customer's case.   So, I added "|| true" to handle the situation.

I wonder if it would make more sense to call the api directly? That sounds more suitable for machine processing. E.g.:
```
curl --header "PRIVATE-TOKEN: ${GITLAB_TOKEN}" "https://gitlab.cee.redhat.com/api/v4/projects/releng%2Fkonflux-release-data/merge_requests?search=Konflux%20release"
```
The response is a json. Then if there is nothing, it's just an empty array. But I'm not sure exactly what UPSTREAM_REPO we can expect here and so we would need to parse that. I guess it's ok to keep it as is for now, but something to consider in the future.

Reply:
It's simple to use `glab` command and `glab` is used the functions in release-service-utils repo. I prefer to use `glab` in the PR.  I know there is potential issue about the MR numbers. If the number of  the returned MRs exceeds 100, it's hard to deal with both with `glab` command and curl command because it's the limitation of gitlab.  Any idea?

Reply:
ok